clients/venice-producer/src/main/java/com/linkedin/venice/producer/VeniceProducer.java

```diff
@@ -0,0 +1,88 @@
+package com.linkedin.venice.producer;
+
+import com.linkedin.venice.writer.update.UpdateBuilder;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
+
+
+/**
+ * The API for online applications to write to Venice. These APIs are only eventually consistent and the futures
+ * returned by them only guarantee that the write operation was durable. There is no option to block until the data
+ * is readable.
+ *
+ * All of these APIs have at least two versions - one that accepts a logical timestamp and one that doesn't.
+ * 1. Logical timestamps (in ms) are what Venice backend will use to resolve conflicts in case multiple writes modify
```

Comment:
> use to resolve conflicts in case multiple writes modify the same record

Is it always last-write-win or case by case?

Reply:
It is mostly LWW but "last" is in terms of the logical timestamp.

For conflicts between nearline updates, it is LWW
For conflicts between batch and nearline, we follow `BatchConflictResolutionPolicy`